uaclient/security.py

```diff
@@ -753,45 +817,74 @@ def _prompt_for_attach(cfg: UAConfig) -> bool:
     return True
 
 
+def _check_subscription_for_required_service(
+    cfg: UAConfig, pocket: str
+) -> bool:
+    """Verify if the ua subscription has the required service enabled."""
+    if pocket == UA_INFRA_POCKET:
+        service_to_check = "esm-infra"
+    else:
+        service_to_check = "esm-apps"
+
+    ent_cls = ENTITLEMENT_CLASS_BY_NAME.get(service_to_check)
+
+    if ent_cls:
+        ent = ent_cls(cfg)
+        ent_status, _ = ent.user_facing_status()
+
+        if ent_status == status.UserFacingStatus.ACTIVE:
+            return True
+
+        print(
+            status.MESSAGE_SECURITY_UA_SERVICE_NOT_ENABLED.format(
```

Comment:
Probably add a TODO comment for a future PR
```suggestion
            # TODO(GH: #1455 prompt when service not enabled)
            status.MESSAGE_SECURITY_UA_SERVICE_NOT_ENABLED.format(
```